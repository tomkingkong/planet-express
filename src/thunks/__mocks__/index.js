export const importSpaceEvents = jest.fn();
export const importScienceNews = jest.fn();
export const importTechnologies = jest.fn();
export const importExoplanets = jest.fn();