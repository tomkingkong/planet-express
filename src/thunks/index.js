import { 
  populateExoplanets, 
  populateTechnologies, 
  populateNews } from '../actions';
import { 
  fetchPlanets, 
  fetchTechnologies, 
  fetchTechnology, 
  fetchScienceNews } from '../utilities/fetches';
import { 
  exoplanetCleaner, 
  techCleaner } from '../utilities/cleaners/';

export const importExoplanets = () => {
  return async (dispatch) => {
    try {
      const exoplanets = await fetchPlanets();
      const limitPlanets = exoplanets.slice(400, 410);
      const cleanPlanets = exoplanetCleaner(limitPlanets);
      dispatch(populateExoplanets(cleanPlanets));
    } catch (error) {
      console.log(error)
    }
  }
}

export const importTechnologies = () => {
  return async (dispatch) => {
    try {
      const technologies = await fetchTechnologies();
      const limitTechnologies = technologies.projects.projects.slice(0, 10);
      const dirtyTechs = await Promise.all(limitTechnologies.map(tech => fetchTechnology(tech.id)));
      const cleanTechs = techCleaner(dirtyTechs);
      dispatch(populateTechnologies(cleanTechs));
    } catch (error) {
      console.log(error)
    }
  }
}

export const importScienceNews = () => {
  return async (dispatch) => {
    try {
      const scienceNews = await fetchScienceNews();
      const newswithImages = await scienceNews.articles.filter(article => article.urlToImage)
      const limitNews = newswithImages.slice(3, 9);
      dispatch(populateNews(limitNews));
    } catch (error) {
      console.log(error)
    }
  }
}