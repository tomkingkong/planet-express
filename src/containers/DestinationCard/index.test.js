import React from 'react';
import { shallow } from 'enzyme';
import { 
  DestinationCard, 
  mapStateToProps, 
  mapDispatchToProps } from '.';
import { createJourney } from '../../thunks/';

describe('DestinationCard Container', () => {
  let wrapper;
  let history;
  let mockPlanet;
  let mockJourneyFn;
  let mockHistoryFn;
  let e;

  beforeEach(() => {
    mockPlanet = {
      name:'tim', 
      image:'butts', 
      distance:'far', 
      numInSystem: 1, 
      hostStar:'tim'
    };
    e = {
      persist: jest.fn(),
      stopPropagation: jest.fn(),
      target: {className: 'confirm__no'}
    };
    mockJourneyFn = jest.fn();
    mockHistoryFn = jest.fn();
    history = { replace: mockHistoryFn };
    wrapper = shallow(
      <DestinationCard 
        history={history}
        spaceEvents={[]} 
        planet={mockPlanet}
        createJourney={mockJourneyFn} 
      />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state of toggleConfirmModal set to false', () => {
    expect(wrapper.state().toggleConfirmModal).toEqual(false);
  });

  it('handleConfirm should switch state of toggleConfirmModal back to false if user declines destination', () => {
    wrapper.instance().handleClick();
    e = {
      persist: jest.fn(),
      stopPropagation: jest.fn(),
      target: `<button class="confirm__no">Decline</button>`
    };
    wrapper.instance().handleConfirm(e);
    expect(wrapper.state().toggleConfirmModal).toEqual(false);
  });

  it('handleConfirm should send the user on to their journey if user accepts destination', () => {
    e = {
      persist: jest.fn(),
      stopPropagation: jest.fn(),
      target: { }
    };
    wrapper.instance().handleConfirm(e);
    expect(mockHistoryFn).toHaveBeenCalled();
  });

  it('handleConfirm should call the createJourney thunk', () => {
    e = {
      persist: jest.fn(),
      stopPropagation: jest.fn(),
      target: { }
    };
    wrapper.instance().handleConfirm(e);
    expect(mockJourneyFn).toHaveBeenCalled();
  });

  describe('mapStateToProps', () => {
    it('should be provided with a spaceEvents array from props', () => {
      const mockState = {
        exoplanets: [],
        spaceEvents: [{}, {}],
        news: [],
        journey: {},
        technologies: []
      };
      const expected = { spaceEvents: [{}, {}] };
      const results = mapStateToProps(mockState);
      expect(results).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with createJourney when createJourney is called', () => {
      const mockDispatch = jest.fn();
      const mockJourey = { "journey": {"encounters": [], "planetId": "id"}, "type": "SET_JOURNEY" };
      const thunk = createJourney([], 'id');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.createJourney([], 'id');
      thunk(mockDispatch);
      expect(mockDispatch).toHaveBeenCalledWith(mockJourey);
    });
  });
});