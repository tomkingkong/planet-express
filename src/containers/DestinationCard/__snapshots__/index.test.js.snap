// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationCard Container should match snapshot 1`] = `
<article
  className="Destination"
  onClick={[Function]}
>
  <div
    className="planet__icon"
  >
    <Component
      planetMap="butts"
    />
  </div>
  <div
    className="destination__info"
  >
    <h4
      className="planet__title"
    >
      tim
    </h4>
    <p>
      far
       parsecs from Earth.
    </p>
    <h5>
      Start your trip!
    </h5>
  </div>
</article>
`;
